go/prometheusexporter/webserver/webserver.go

```diff
@@ -11,10 +11,13 @@ import (
 	"gopkg.in/tylerb/graceful.v1"
 )
 
+type serveMetrics func(w http.ResponseWriter, r *http.Request)
+
 // HandleHTTPRequests starts a webserver for exposing prometheus metrics.
 func HandleHTTPRequests(port string) {
 	r := mux.NewRouter()
 	r.HandleFunc("/metrics/{application}/{environment}", serveAppMetrics)
+	r.HandleFunc("/metrics/{application}/{environment}/exceptions", serveExceptionsMetrics)
```

Comment:
I think you need to reverse the registration order here, otherwise the first handler will be always called. Personally I would probably have performed the test inside the serveAppMetrics function instead.

Reply:
I don't think that this would be necessary. The docs contain a very similar example: https://github.com/gorilla/mux#matching-routes

```
s.HandleFunc("/products/", ProductsHandler)
s.HandleFunc("/products/{key}", ProductHandler)
s.HandleFunc("/articles/{category}/{id:[0-9]+}", ArticleHandler)
```

The docs also state that variables only match until the next `/` (see [here](https://pkg.go.dev/github.com/gorilla/mux#Route.Path)).

> Path adds a matcher for the URL path. It accepts a template with zero or more URL variables enclosed by {}. The template must start with a "/". Variables can define an optional regexp pattern to be matched:

- {name} matches anything until the next slash.

- {name:pattern} matches the given regexp pattern.

Reply:
Last time I looked at the source code, the route is found by the first match in a list of matchers. There is no compilation into a trie or anything to find the largest matching path part. But maybe I have overlooked something.

Reply:
Maybe it returns the last match instead of the first.

Reply:
I guess they do some magic with string ends boundaries.